sniffles/test/test_feature.py

```diff
@@ -0,0 +1,310 @@
+from unittest import *
+from sniffles.feature import *
+import random
+
+
+class TestFeature(TestCase):
+    def test_ambigous_notation_class(self):
+        ambigous = AmbiguousNotation("[5,3,8]")
+        self.assertEqual(ambigous.toString(), "[5,3,8]")
+
+    def test_set_notation_class(self):
+        test = SetNotation("{mon,tues,wed,thurs,fri,sat,sun}")
+        testList = ["mon", "tues", "wed", "thurs", "fri", "sat", "sun"]
+        for i in range(0, 10):
+            values = (test.toString()[1:-1]).split(",")
+            for value in values:
+                self.assertTrue(value in testList)
+
+    def test_range_notation_class(self):
+        for i in range(0, 5):
+            lower = random.randint(0, 20)
+            upper = random.randint(21, 25)
+            test = RangeNotation("[" + str(lower) + ":" +
+                                 str(upper) + "]")
+            result = test.toString()
+            myrange = result[1:-1]
+            bounds = myrange.split(":")
+            lower = int(bounds[0])
+            upper = int(bounds[1])
+            self.assertTrue(lower >= 0)
+            self.assertTrue(upper <= 25)
+            self.assertTrue(lower <= upper)
+
+    def test_list_notation_class(self):
+        # basic test case [5,6]
+        test1 = ListNotation("[5,6]")
+        self.assertEqual(str(test1), "[5,6]")
+
+        # # every value in the list should be in
+        # # range [5, 150]
+        test2 = ListNotation("[5,150]")
+        mySplit = test2.toString()[1:-1]
+        values = mySplit.split(",")
+        for value in values:
+            iVal = int(value)
+            self.assertTrue(iVal >= 5)
+            self.assertTrue(iVal <= 150)
+
+        # [5,4] --> [3,4]
+        test3 = ListNotation("[5,4]")
+        self.assertEqual(str(test3), "[3,4]")
+
+        # [5,0] --> [0,1]
+        test3 = ListNotation("[5,0]")
+        self.assertEqual(str(test3), "[0,1]")
+
+    def test_feature_class(self):
+        # no ambiguity list
+        myfeature = Feature("f1", 20, 30, 0)
```

Comment:
rather than parsing the to string function, maybe we should add proper getters to the feature class.  In other words, why don't we add getName(), getLowerBound(), etc to class Feature.